lib/matplotlib/units.py

```diff
@@ -114,61 +115,28 @@ def is_numlike(x):
 
 
 class Registry(dict):
-    """
-    register types with conversion interface
-    """
-    def __init__(self):
-        dict.__init__(self)
-        self._cached = {}
+    """Register types with conversion interface."""
 
     def get_converter(self, x):
-        'get the converter interface instance for x, or None'
-
-        if not len(self):
-            return None  # nothing registered
-        # DISABLED idx = id(x)
-        # DISABLED cached = self._cached.get(idx)
-        # DISABLED if cached is not None: return cached
-
-        converter = None
-        classx = getattr(x, '__class__', None)
-
-        if classx is not None:
```

Comment:
Shouldn't this also stay in, as there's no guarantee that unit data is a subclass of numpy array or an iterable?

Reply:
Should be handled by `return self[type(x)]`, no?

Reply:
Ah yes